### PR TITLE
Gc(test bot)

### PR DESCRIPTION
**Pull request by likeRewca**

### CODE REVIEW

**Comment by likeRewca** [comment]
Some comment

**Comment by likeRewca** [comment]
One more comment

File: module1/src/main/java/net/broscorp/gc/GcTest.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    this.numberOfObject = countOfObject++;
+  }
+
+  /**
+   * Создаем большое количество объектов и сразу теряем ссылку.
+   */
+  public void largeNumberOfObjects() {
+    for (int i = 0; i < 1_000_000; i++) {
+      GcTest gcTest = new GcTest("Object");
+    }
```
Comment (likeRewca) [inline module1/src/main/java/net/broscorp/gc/GcTest.java:27]: Some annotation for bot

File: module1/src/main/java/net/broscorp/gc/GcTest.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+  //  @Override
+  //  protected void finalize() {
+  //    try (FileWriter writer = new FileWriter("secondLog.txt", true)) {
+  //      String massage = this.name + " - " + this.numberOfObject + " was delete.";
+  //      writer.append(massage).append("\n");
+  //      writer.flush();
+  //    } catch (IOException ex) {
+  //      System.out.println(ex.getMessage());
+  //    }
```
Comment (likeRewca) [inline module1/src/main/java/net/broscorp/gc/GcTest.java:56]: Don`t do comment!!

File: module1/src/main/java/net/broscorp/gc/Immortal.java [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+import java.util.HashSet;
+import java.util.Set;
+
+public class Immortal {
+  private static final Set<Immortal> immortals = new HashSet<>();
+
+  //  @Override
+  //  protected void finalize() throws Throwable {
+  //    System.out.println(Immortal.class.getSimpleName() + "::finalize for " + this);
+  //    immortals.add(this);
```
Comment (likeRewca) [inline module1/src/main/java/net/broscorp/gc/Immortal.java:12]: the same

**Review by likeRewca — COMMENTED** [review]
Come check

File: module1/src/main/java/net/broscorp/gc/GcTest.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+  private GcTest gc;
+  private static int countOfObject = 0;
+  private int numberOfObject;
+
+  public GcTest() {
+  }
+
+  public GcTest(String name) {
+    this.name = name;
+    this.numberOfObject = countOfObject++;
```
Comment (likeRewca) [inline module1/src/main/java/net/broscorp/gc/GcTest.java:18]: вфжыв

File: module1/src/main/java/net/broscorp/gc/GcTest.java [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+  public void setGc(GcTest gcTest) {
+    this.gc = gcTest;
+  }
+
+  //  @Override
+  //  protected void finalize() {
+  //    try (FileWriter writer = new FileWriter("secondLog.txt", true)) {
+  //      String massage = this.name + " - " + this.numberOfObject + " was delete.";
+  //      writer.append(massage).append("\n");
```
Comment (likeRewca) [inline module1/src/main/java/net/broscorp/gc/GcTest.java:52]: вывыв

**Comment by likeRewca** [comment]
review

File: module1/src/main/java/net/broscorp/gc/Test.java [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    //new Immortal();
+    //System.gc();
+    //sleep(1_000);
+    //System.gc();
+    //prompt("Press any key...");
+  }
+
+  private static void prompt(String msg) {
+    try {
+      System.out.println(msg);
```
Comment (likeRewca) [inline module1/src/main/java/net/broscorp/gc/Test.java:24]: Good one

File: module1/src/main/java/net/broscorp/gc/Test.java [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+      System.in.read();
+    } catch (IOException io) {
+      System.out.println("some trouble");
+    }
+  }
+
+  private static void sleep(long duration) {
+    try {
+      Thread.sleep(duration);
```
Comment (likeRewca) [inline module1/src/main/java/net/broscorp/gc/Test.java:33]: Good two

**Review by likeRewca — COMMENTED** [review]
Approved

**Comment by likeRewca** [comment]
Approved

File: module1/src/main/java/net/broscorp/gc/GcTest.java (hunk 1 above)
Comment (marvintik) [reply module1/src/main/java/net/broscorp/gc/GcTest.java:27]: Some annotation

File: module1/src/main/java/net/broscorp/gc/GcTest.java (hunk 5 above)
Comment (marvintik) [reply module1/src/main/java/net/broscorp/gc/GcTest.java:52]: Leave a comment

File: module1/src/main/java/net/broscorp/gc/GcTest.java (hunk 1 above)
Comment (marvintik) [reply module1/src/main/java/net/broscorp/gc/GcTest.java:27]: approved

**Comment by marvintik** [comment]
Approved

**Comment by marvintik** [comment]
Approved

**Pull request closed without merging**